src/System.Private.CoreLib/shared/System/Nullable.cs

```diff
@@ -71,10 +71,12 @@ public override int GetHashCode()
             return hasValue ? value.GetHashCode() : 0;
         }
 
+#pragma warning disable CS8609 // TODO-NULLABLE: Covariant return types (https://github.com/dotnet/roslyn/issues/23268)
         public override string? ToString()
```

Comment:
When is this non-null? `value` is struct and `ValueType.ToString` is annotated as non-null.

Reply:
Is the difference between `Object.ToString` and `ValueType.ToString` intentional? It looks like one of those things that might have been discussed extensively - it would be nice to capture the key point of the discussion as a comment on `ValueType.ToString`

Reply:
> Is the difference between Object.ToString and ValueType.ToString intentional?

It is not.  `ValueType.ToString` should also be `string?`.  There are a non-trivial number of cases we found in many projects where ToString just returns some field of a struct where that field may be null, e.g. https://github.com/dotnet/roslyn/blob/51af99e253c21e268518feb6a1fd4e123240fd70/src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayPart.cs#L60

Reply:
I agree. I missed this one as I was just disabling the warns that fall back from removing the NoWarn globally. Actually I annotated ValueType and `ValueType.ToString` was annotated as returning nullable, but it seems that in the PR to annotate `System.Reflection` it was reverted to return non-null.

https://github.com/dotnet/coreclr/commit/56f38ea2a930b18b0c43f717f90496b44a5d9283

Reply:
Fixed.

Reply:
I still think that making ToString string? is one of the biggest mistakes there can be. I've yet to see a place where a ToString returns null and that is not an implementation error/mistake but would actually make sense.
Our code has (literally) tens of thousands of ToString calls and overloads and there is no single place where a null would be handled (or should be handled).

Reply:
> Our code has (literally) tens of thousands of ToString calls

On `object` or on one of your types that overrides `ToString`?  For cases where it's the former, what is most commonly done with the result?